lib/oban/notifier.ex

```diff
@@ -84,232 +22,57 @@ defmodule Oban.Notifier do
+  @doc false
+  @spec unlisten(server(), [channel]) :: :ok
+  def unlisten(server, channels) do
+    conf = Oban.config(server)
 
-      Oban.Notifier.unlisten(MyApp.Oban, [:gossip, :signal])
-  """
-  @spec unlisten(GenServer.server(), channels :: list(channel())) :: :ok
-  def unlisten(oban_name \\ Oban, [_ | _] = channels) do
-    oban_name
-    |> Registry.whereis(__MODULE__)
-    |> GenServer.call({:unlisten, channels})
+    server
+    |> Registry.whereis(Oban.Notifier)
+    |> conf.notifier.unlisten(channels)
   end
 
-  @doc """
-  Broadcast a notification to listeners on all nodes.
-
-  Notifications are scoped to the configured `prefix`. For example, if there are instances running
-  with the `public` and `private` prefixes, a notification published in the `public` prefix won't
-  be picked up by processes listening with the `private` prefix.
-
-  ## Example
-
-  Broadcast a gossip message:
-
-      Oban.Notifier.notify(Oban.config(), :gossip, %{message: "hi!"})
-  """
+  @doc false
   @spec notify(Config.t(), channel :: channel(), payload :: map() | [map()]) :: :ok
-  def notify(%Config{} = conf, channel, %{} = payload) when is_channel(channel) do
-    notify(conf, channel, [payload])
-  end
-
-  def notify(%Config{} = conf, channel, [_ | _] = payload) when is_channel(channel) do
-    channel = "#{conf.prefix}.#{@mappings[channel]}"
-
-    Repo.query(
-      conf,
-      "SELECT pg_notify($1, payload) FROM json_array_elements_text($2::json) AS payload",
-      [channel, Enum.map(payload, &Jason.encode!/1)]
-    )
-
-    :ok
-  end
-
-  @impl GenServer
-  def init(opts) do
-    Process.flag(:trap_exit, true)
-
-    {:ok, struct!(State, opts), {:continue, :start}}
-  end
-
-  @impl GenServer
-  def handle_continue(:start, state) do
-    {:noreply, connect_and_listen(state)}
-  end
-
-  @impl GenServer
-  def handle_info({:DOWN, _ref, :process, pid, _reason}, %State{listeners: listeners} = state) do
-    {:noreply, %{state | listeners: Map.delete(listeners, pid)}}
-  end
-
-  def handle_info({:notification, _, _, prefixed_channel, payload}, state) do
-    full_channel =
-      prefixed_channel
-      |> String.split(".")
-      |> List.last()
-
-    channel = reverse_channel(full_channel)
-    decoded = if any_listeners?(state.listeners, full_channel), do: Jason.decode!(payload)
-
-    if in_scope?(decoded, state.conf) do
-      for {pid, channels} <- state.listeners, full_channel in channels do
-        send(pid, {:notification, channel, decoded})
-      end
-    end
-
-    {:noreply, state}
-  end
-
-  def handle_info({:EXIT, _pid, error}, %State{} = state) do
-    state = trip_circuit(error, [], state)
-
-    {:noreply, %{state | conn: nil}}
-  end
-
-  def handle_info(:reset_circuit, %State{circuit: :disabled} = state) do
-    state =
-      state
-      |> open_circuit()
-      |> connect_and_listen()
-
-    {:noreply, state}
+  def notify(%Config{} = conf, channel, payload) when is_channel(channel) do
```

Comment:
Removing this config would introduce a compatibility break. Shall I?

Reply:
We can soft-deprecate the old version and transparently map it to the new one.